Project 5 For John Beverly

```diff
@@ -59,345 +59,388 @@ Group 2 contains Emanuel, Katie, Josh V (commonly know as V by group member), Ca
 
-sh:property [        
+sh:property [    
 
-sh:path schema:name ;        
+sh:path ex:hasSource ;    
 
-sh:datatype xsd:string ;       
+sh:class ex:Node 
 
-sh:minLength 3 ;       
+] ;
 
-sh:message "The album name must be a string of at least 3 characters."   
+sh:property [    
 
-] ;
 
-sh:property [       
+sh:path ex:hasTarget ;    
 
-sh:path schema:byArtist ;        
+sh:class ex:Node  
 
-sh:node ex:ArtistShape ;        
+] ;
+
+sh:property [    
 
-sh:message "The album artist must be a valid artist object."    ] ;
+sh:path ex:hasValue ;    
 
-sh:property [       
+sh:datatype xsd:boolean ;    
 
-sh:path schema:datePublished ;        
+sh:in ( true )  ] ;
 
-sh:datatype xsd:date ;        
+sh:property [    
 
-sh:message "The album release date must be a valid date in the format YYYY-MM-DD."    ] ;
+sh:path ex:hasType ;    
 
+sh:in ( "asymmetric" )  
 
-sh:property [        
+] .
 
-sh:path schema:numTracks ;        
+9.	Constraint for disjoint of symmetry and asymmetry 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-sh:datatype xsd:integer ;        
+@prefix ex: <http://example.com/> .
 
-sh:minInclusive 1 ;        
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:message "The album must have at least 1 track."   
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-] ;
+ex:symmetricalPropertyShape
 
-sh:property [       
+a sh:PropertyShape ;
 
-sh:path schema:recordLabel ;       
+sh:path ex:symmetricalProperty ;
 
-sh:datatype xsd:string ;       
+sh:propertyConstraint [
 
-sh:message "The album record label must be a string."   
+sh:symmetrical true ;
 
 ] .
 
 
-# Define the shape for the artist object
+ex:asymmetricalPropertyShape
+
+a sh:PropertyShape ;
 
-ex:ArtistShape
+sh:path ex:asymmetricalProperty ;
+
+sh:propertyConstraint [
+
+sh:asymmetrical true ;
+
+] .
+
+ex:MyClassShape
 
 a sh:NodeShape ;
 
-sh:targetClass schema:MusicGroup ;
 
-sh:property [        
+sh:property [
 
-sh:path schema:name ;        
+sh:xoneOf ( ex:symmetricalPropertyShape ex:asymmetricalPropertyShape ) ;
 
-sh:datatype xsd:string ;       
+] .
 
-sh:minLength 3 ;        
 
-sh:message "The artist name must be a string of at least 3 characters."   
 
-] ;
-    sh:property [        
+10.	Constraint for PropertyIn for Asymmetrical and Irrelexive 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-sh:path schema:genre ;        
+@prefix ex: <http://example.com/> .
 
-sh:datatype xsd:string ;       
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:message "The artist genre must be a string."   
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-] .
 
+ex:asymmetricalPropertyShape
 
-# Define the shape for the user object
+a sh:PropertyShape ;
 
-ex:UserShape
+sh:path ex:asymmetricalProperty ;
 
-a sh:NodeShape ;
+sh:propertyConstraint [
 
-sh:targetClass schema:Person ;
+sh:asymmetrical true ;
 
-sh:property [        
+] .
 
-sh:path schema:name ;        
+ex:irreflexivePropertyShape
```

Comment:
So you are giving to shapes here, right?

Reply:
Not quite, and again I am pretty shaky on this but per my understanding the place of ex:irreflexivePropertyShape here is being used as a path value for the property. The shape definition itself is the sh:irreflexive property being set to true. such that the ClassShape uses the asymmeticalPropertyShape and irreflexivePropertyShape in its disjunct for validating the overall shape. 
I am not 100% I described this right or intelligibly but I hope I did and that this helps :)